src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -178,6 +178,7 @@ export class MockHttpSocket extends MockSocket {
 
     const socket = this.createConnection()
     this.originalSocket = socket
+    this._handle = { reading: true, close: () => {} }
```

Comment:
Why not infer the `_handle` of the original `socket` since we are in a passthrough context anyway? What do we gain by keeping a dummy socket, save that it passes the `this._handle` internal Node.js checks?

I would image we should do 

```ts
this._handle = socket._handle
```

One caution here is taking a look at _when_ the handle is assigned. Afaik, that happens in the Socket constructor, but in case it happens after, we should listen to that after and assign the `_handle` value then. 

What do you think?

Reply:
It's working, but I can't remember why I didn't take the `socket._handle`. I think I was afraid that the mocked socket and the real socket would somehow interfere with each other.
Anyway, it's working, and all the tests pass, so I'm okay to do this and fix any errors that occur.

Reply:
In this context, we are only handling passthrough requests so mocked functionality mustn't interfere with them. It should be safe to get the `_handle` create by Node.js.